gmso/core/atom.py

```diff
@@ -0,0 +1,132 @@
+import warnings
+from typing import Optional, Union
+
+import unyt as u
+
+from pydantic import Field, validator
+
+from gmso.core.element import Element
+from gmso.abc.abstract_site import Site
+from gmso.core.atom_type import AtomType
+
+
+class Atom(Site):
+    __base_doc__ = """An atom represents a single element association in a topology.
+    Atom is the object that represents any general interaction Atom in a molecular simulation.
+
+    Notes
+    -----
+    Atoms have all the attributes inherited from the base Site class
+
+    Examples
+    --------
+        >>> from gmso.core.atom import Atom
+        >>> atom1 = Atom(name='lithium')
+
+    See Also
+    --------
+    gmso.abc.AbstractSite
+        An Abstract Base class for implementing site objects in GMSO. The class Atom bases from
+        the gmso.abc.abstract site class
+    """
+    charge_: Optional[Union[u.unyt_quantity, float]] = Field(
+        None,
+        description='Charge of the atom',
+    )
+
+    mass_: Optional[Union[u.unyt_quantity, float]] = Field(
```

Comment:
unyt_quantities/unyt_arrays are not json serializable right now. That's probably not in the scope of this PR, but it is holding up our ability to serialize Atoms (and other classes) at the moment.

Reply:
NVM you mentioned this in a comment earlier